pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -894,5 +895,21 @@ func (lb *ListenerBuilder) buildInboundNetworkFilters(fcc inboundChainConfig) []
 	filters = append(filters, buildMetricsNetworkFilters(lb.push, lb.node, istionetworking.ListenerClassSidecarInbound)...)
 	filters = append(filters, buildNetworkFiltersStack(fcc.port.Protocol, tcpFilter, statPrefix, fcc.clusterName)...)
 
+	sort.SliceStable(filters, func(i, j int) bool {
```

Comment:
why we are sorting after the fact instead of just inserting them in the correct order to begin with?

Reply:
i considered that.  this approach felt less error-prone, stems mostly from my lack of experience with the istio codebase.  i am hoping an istio committer can revise the implementation to precisely what you suggest.